enterprise/admin/forms.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+"""
+Custom admin forms.
+"""
+from __future__ import absolute_import, unicode_literals
+
+from django import forms
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+from django.core.validators import validate_email
+from django.utils.translation import ugettext as _
+
+from enterprise.models import EnterpriseCustomerUser
+
+
+class ManageLearnersForm(forms.Form):
+    """
+    Form to manage learner additions.
+    """
+    email = forms.CharField(label=_("Email or Username"))
+
+    def clean_email(self):
+        """
+        Clean email form field
+        :return: str email
+        """
+        email_or_username = self.cleaned_data['email']
+        try:
+            user = User.objects.get(username=email_or_username)
+            email = user.email
+        except User.DoesNotExist:
+            email = email_or_username
+
+        try:
+            validate_email(email)
+        except ValidationError:
+            message = _("{email_or_username} does not appear to be a valid email or known username").format(
+                email_or_username=email
+            )
+            raise ValidationError(message)
+
+        try:
+            existing_record = EnterpriseCustomerUser.objects.get(user_email=email)
```

Comment:
@mtyaka As far as I can tell it is not supported. @mattdrayer or am I missing something?